kotest-property/kotest-property-core/src/commonMain/kotlin/io/kotest/property/core/proptest.kt

```diff
@@ -0,0 +1,59 @@
+package io.kotest.property.core
+
+import io.kotest.mpp.sysprop
+import io.kotest.property.AssumptionFailedException
+import io.kotest.property.Gen
+import io.kotest.property.PropertyTesting
+import io.kotest.property.RandomSource
+import io.kotest.property.ShrinkingMode
+import kotlin.random.Random
+import kotlin.time.Duration
+
+class PropTestContext {
+
+   var iterations: Int? = null // converted to iteration based constraint
+   var duration: Duration? = null // converted to duration based constraint
+   var constraints: Constraints? = null // specify manual constraits
+   var minSuccess = -1
+   var maxFailure = -1
+   var edgecaseFactor = PropertyTesting.defaultEdgecasesGenerationProbability
+   var shouldPrintShrinkSteps = false // sysprop("kotest.proptest.output.shrink-steps", true)
+   var shrinkingMode: ShrinkingMode = PropertyTesting.defaultShrinkingMode
+   var maxDiscardPercentage: Int = 20
+   var shouldPrintGeneratedValues: Boolean = PropertyTesting.shouldPrintGeneratedValues
+
+   // must call seed before rs otherwise has no effect
+   var seed: Long? = null
+   var failOnSeed: Boolean = sysprop("kotest.proptest.seed.fail-if-set", false)
+
+   internal val rs = lazy { RandomSource.seeded(seed ?: Random.nextLong()) }
+
+   internal val container = GenDelegateContainer()
+
+   internal var property: suspend () -> Unit = {}
+   internal var beforeProperty: suspend () -> Unit = {}
+   internal var afterProperty: suspend () -> Unit = {}
```

Comment:
I regularly encountered situations that required me to clear mocks, also mocks used by the whole spec, after every iteration; it might be nice to have this on a spec-level, for all test & property-tests (like `beforeTestIteration`, which would be called before a "normal" test and before any property test iteration).

Reply:
Maybe a combined beforeIteration could do that. The problem is I didn't really want to have property testing as part of kotlin framework directly, but maybe we should just do that.